tests/unit_test/private/fed/app/simulator/simulator_runner_test.py

```diff
@@ -86,3 +87,34 @@ def test_split_names(self, client_names, gpus, expected_split_names):
         runner = SimulatorRunner(job_folder="", workspace="")
         split_names = runner.split_clients(client_names, gpus)
         assert sorted(split_names) == sorted(expected_split_names)
+
+    @pytest.mark.parametrize(
+        "gpus, expected_gpus",
+        [
+            ("[0,1],[1, 2]", ["0,1", "1,2"]),
+            ("[0,1],[3]", ["0,1", "3"]),
+            ("[0,1],[ 3 ]", ["0,1", "3"]),
+            ("[02,1],[ a ]", ["02,1", "a"]),
+            ("[]", [""]),
+            ("[0,1],3", ["0,1", "3"]),
+            ("0,1,2", ["0", "1", "2"]),
+        ],
```

Comment:
Suggest to add some negative cases:

"[2,1], ]1,2,3["
"[1,3,4"

Reply:
All these negative cases already handled, and the unit tests covered in test_split_gpus_fail()